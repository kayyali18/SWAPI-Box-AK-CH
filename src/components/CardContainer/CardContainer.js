import React, { Component } from "react"
import PropTypes from "prop-types"

import Card from "../Card/Card"

const CardContainer = (props) => {
console.log(props.cards)
  if (props.cards) {

  const cards = props.cards.map((card, index) => {
      return (
        <Card
          data={card}
          key={index}
          category={props.category}
        />
        )
      })

    return (
      <section className="l-card-container card-container">
        { cards }
      </section>
    )
  } else {
    return null
  }
}

CardContainer.propTypes = {
  people: PropTypes.object,
  planets: PropTypes.object,
  vehicles: PropTypes.object
}

export default CardContainer;