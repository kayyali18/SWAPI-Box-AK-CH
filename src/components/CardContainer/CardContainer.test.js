import React from 'react'
import ReactDOM from 'react-dom'
import CardContainer from './CardContainer'
import {shallow, configure} from 'enzyme'

describe('Card Container', () => {
  let wrapper
  let mockProps

  beforeEach(() => {
    wrapper = shallow(<CardContainer />)
  })

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("should fetch supplementary data and update state", () => {

  })
})