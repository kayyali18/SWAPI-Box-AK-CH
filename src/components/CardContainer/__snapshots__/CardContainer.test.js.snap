// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Container renders without crashing 1`] = `""`;
